src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -45,6 +48,64 @@ public TelemetryRecorder(IFileSystemUtils fileSystemUtils, IConfiguration config
         Log = log ?? throw new ArgumentNullException(nameof(log));
     }
 
+    private TelemetryRecorder(IConfiguration configuration, IFileSystemUtils fileSystemUtils)
+    {
+        Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        FileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+    }
+
+    /// <summary>
+    /// Create an instance of TelemtryRecorder without the need for an ILogger.
+    /// </summary>
+    /// <param name="configuration">Current configuration of the tool.</param>
+    /// <param name="fileSystemUtils">Wrapper around file system functions.</param>
+    public static TelemetryRecorder Create(IConfiguration configuration, IFileSystemUtils fileSystemUtils)
+    {
+        return new TelemetryRecorder(configuration, fileSystemUtils);
+    }
+
+    /// <summary>
+    /// Method to log telemetry in conditions when the tool is not able to start execution of workflow.
+    /// </summary>
+    /// <param name="exception">Exception that we want to log.</param>
+    public async Task LogToConsole(Exception exception)
```

Comment:
Looks like you are not only logging to Console, should this method be named something like LogException, instead?